source/slang/slang-ir-autodiff-fwd.cpp

```diff
@@ -955,20 +955,22 @@ InstPair ForwardDiffTranscriber::transcribeCall(IRBuilder* builder, IRCall* orig
 InstPair ForwardDiffTranscriber::transcribeSwizzle(IRBuilder* builder, IRSwizzle* origSwizzle)
 {
     IRInst* primalSwizzle = maybeCloneForPrimalInst(builder, origSwizzle);
-
     if (auto diffBase = lookupDiffInst(origSwizzle->getBase(), nullptr))
     {
-        List<IRInst*> swizzleIndices;
-        for (UIndex ii = 0; ii < origSwizzle->getElementCount(); ii++)
-            swizzleIndices.add(origSwizzle->getElementIndex(ii));
+        if (auto diffType = differentiateType(builder, primalSwizzle->getDataType()))
```

Comment:
Can you explain here why `diffType` maybe null when `diffBase` is not null?

Reply:
Added a comment 👍